File: Frontend/SmartSolarMicrogrid.Web/src/types/bookingSlot.ts
export interface BookingSlot {
  id?: string;
  slotId: string;
  stationId: string;
  startTime: string;
  endTime: string;
  totalCapacity: number;
  availableCapacity: number;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export interface BookingSlotsResponse {
  slots: BookingSlot[];
}

export interface CreateBookingSlotRequest {
  stationId: string;
  startTime: string;
  endTime: string;
  totalCapacity: number;
}

export interface UpdateBookingSlotRequest {
  startTime: string;
  endTime: string;
  totalCapacity: number;
}

File: Android/app/src/main/java/com/smartsolar/microgrid/model/BookingSlot.kt
package com.smartsolar.microgrid.model

/**
 * Represents an energy booking slot published by a station.
 */
data class BookingSlot(

    val id: String? = null,

    val slotId: String,

    val stationId: String,

    val startTime: String,

    val endTime: String,

    val totalCapacity: Int,

    val availableCapacity: Int,

    val isActive: Boolean,

    val createdAt: String? = null,

    val updatedAt: String? = null,
)
